plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java

```diff
@@ -27,19 +28,25 @@
 import io.trino.plugin.jdbc.credential.CredentialProvider;
 import org.postgresql.Driver;
 
+import static io.airlift.configuration.ConditionalModule.installModuleIf;
 import static io.airlift.configuration.ConfigBinder.configBinder;
 import static io.trino.plugin.jdbc.JdbcModule.bindSessionPropertiesProvider;
 
 public class PostgreSqlClientModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     @Override
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
         binder.bind(JdbcClient.class).annotatedWith(ForBaseJdbc.class).to(PostgreSqlClient.class).in(Scopes.SINGLETON);
         configBinder(binder).bindConfig(PostgreSqlConfig.class);
         bindSessionPropertiesProvider(binder, PostgreSqlSessionProperties.class);
-        binder.install(new DecimalModule());
+        install(new DecimalModule());
+
+        install(installModuleIf(
+                PostgreSqlConfig.class,
+                PostgreSqlConfig::isRemoteQueryCancellationEnabled,
+                CachedThreadPoolProvider.bindForRecordCursor()));
```

Comment:
We don't need a Provider for that, since the can be easily achieved in Module

```
install(installModuleIf(
        PostgreSqlConfig.class,
        PostgreSqlConfig::isRemoteQueryCancellationEnabled,
        binder1 -> {
            newOptionalBinder(binder1, Key.get(ExecutorService.class, ForRecordCursor.class))
                    .setBinding()
                    .toProvider(() -> newCachedThreadPool(daemonThreadsNamed(catalogName + "record-cursor-%d")))
                    .in(Scopes.SINGLETON);
        }));
```

I understand is't not a oneliner, so we may want to extract the common part somewhere.
I'd see it more as a Module than a Provider, or a Guice-related helper method.

Reply:
(nvrm anyway)